src/dialects/abstract/query-generator.js

```diff
@@ -2218,10 +2218,24 @@ export class AbstractQueryGenerator extends AbstractQueryGeneratorTypeScript {
     throw new sequelizeError.QueryError(message.replace(/ +/g, ' '));
   }
 
+  _validateSelectOptions(options) {
+    if (options.maxExecutionTimeHintMs != null && !this.dialect.supports.maxExecutionTimeHint.select) {
+      throw new Error(`The maxExecutionTimeMs option is not supported by ${this.dialect.name}`);
+    }
+  }
+
+  _getSelectOptimizerHints(_options) {
+    return 'SELECT';
+  }
+
   selectFromTableFragment(options, model, attributes, tables, mainTableAs) {
     this._throwOnEmptyAttributes(attributes, { modelName: model && model.name, as: mainTableAs });
 
-    let fragment = `SELECT ${attributes.join(', ')} FROM ${tables}`;
+    this._validateSelectOptions(options);
+
+    let fragment = this._getSelectOptimizerHints(options);
```

Comment:
I would actually keep SELECT here, it's not part of the hint

```suggestion
    let fragment = 'SELECT ' + this._getSelectOptimizerHints(options);
```

`_getSelectOptimizerHints` can return `''` instead

Reply:
I think once this is done, we're good to merge :)

Reply:
In that case, we should branch in abstract query generator to make query for MySQL and MariaDB seperately like below.
```
let fragment = 'SELECT';
if (this.dialect.name === 'mysql') {
  fragment = fragment + this._getSelectOptimizerHints(options);
} else if (this.dialect.name === 'mariadb') {
  fragment = this._getSelectOptimizerHints(options) + fragment;
} else {
  ...
}
```
Because (I said in [above conversation](https://github.com/sequelize/sequelize/pull/15341#discussion_r1038752650)) MariaDb and MySQL are have difference to make optimizer hint - MySQL's optimizer hint must be located after SELECT(`SELECT /*+ MAX_EXECUTION_TIME(N) */ FROM ~`) but MariaDB's must before SELECT(`SET STATEMENT max_statement_time=100 FOR SELECT`).

The reason I'm reluctant to branch from abstract query generators is that I believe abstracts shouldn't be materialized. What do you think? @ephys